File: 27-remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int j,k;

        if(nums.size()==0) {
            return 0;
        }else {
            for(int i=nums.size()-1; i>=0; i--) {
                if(nums[i]!=val) {
                    k = i;
                    j = i-1;
                    break;
                }
            }

            for(; j>=0; j--) {
                if(nums[j] == val) {
                    swap(nums[j], nums[k]);
                    k--;
                }
            }

            return nums.size() - count(nums.begin(), nums.end(), val);
            
        }
    }
};

File: 27-remove-element/remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        nums.sort()
        j = len(nums)-1
        if j+1==0:
            return 0
        else:
            for i in range(len(nums)):
                if i>=j:
                  break
                if nums[i] == val:
                    tmp = nums[i]
                    nums[i] = nums[j]
                    nums[j] = tmp
                    j -= 1
            
            return len(nums) - nums.count(val)
